homeassistant/components/alarm_control_panel/__init__.py

```diff
@@ -52,13 +59,24 @@ def setup(hass, config):
     def alarm_service_handler(service):
         """Map services to methods on Alarm."""
         target_alarms = component.extract_from_service(service)
-
         code = service.data.get(ATTR_CODE)
 
-        method = SERVICE_TO_METHOD[service.service]
```

Comment:
@Cinntax Ok could do, did think it might not be OK the way I had done - just though it made code clearer to understand what was happening with the if statements
@MartinHjelmare Cover component seems to handle it well - I don't understand how it it passes the appropriate parameters though? Any tips for understanding that?
When I tried it like that in Envisalink it kept passing code to the keypress parameter which wasn't helpful.